operations/helm/charts/grafana-agent/values.yaml

```diff
@@ -189,6 +189,9 @@ controller:
   # -- Whether to enable automatic deletion of stale PVCs due to a scale down operation, when controller.type is 'statefulset'.
   enableStatefulSetAutoDeletePVC: false
 
+  # -- The StatefulSetAutoDeletePVC feature gate went into alpha status with k8s 1.27. If you have it enabled on an earlier version, you can lower this value as far as 1.23.
+  minK8sVersionForAutoDelete: ">= 1.27"
```

Comment:
I'm looking at what other Helm charts do here to remove the need for another option. It looks like the Loki chart [only checks for 1.23](https://github.com/grafana/loki/blob/5c502e497fcf1e53030a885046098ee9981bf07b/production/helm/loki/templates/read/statefulset-read.yaml#L27). I see a similar check in the [Consul Helm chert](https://github.com/hashicorp/consul-k8s/blob/5ed8514254e026f562d703e2acc7ae7595ad6610/charts/consul/templates/server-statefulset.yaml#L47).

Ideally we'd avoid a new config option here, and rely on documentation on `enableStatefulSetAutoDeletePVC`, where we mention it's a feature gate from 1.23 to 1.26, and generally available afterwards.